kubernetes/kops
# release 1.25.0

- Fix openstack tag limitation
- Add support for installing dcgm exporter
- support for scaleway in s3 buckets
- AWS IAM Role listing: don't ignore "other" errors
- Warn that enabling irsa can be disruptive
- Fix etcd administration link
- Update dependencies
- Remove warning for FindClusterStatus not implemented for Hetzner
- Update Calico to v3.24.1
- Update Canal to v3.24.1
- Run hack/update-expected.sh
- Update Flannel to v0.19.2
- Run hack/update-expected.sh
- Add support for cert-manager dns-01 challenges
- Don't add add IAM vars to manifest if service account is not being created
- Release notes for 1.23.4
- Release notes for 1.24.2
- Makefile: Don't assume GOBIN is set
- GCE: change default control-plane instance type to e2-medium
- applylib: Better health checking
- Bump cluster-autoscaler images
- Run hack/update-expected.sh
- Small notes cleanup
- Make 1.25 notes visible in the side bar
- Add support to --cordon-node-before-terminating on the cluster autoscaler addon (CordonNodeBeforeTerminating)
- Bump versions in netlify and mkdocs
- aws-node-termination-handler: Add option to fetch node name through Kubernetes node `.spec.providerID`
- Update dependencies
- Warm pool-enabled ASGs scaled to zero will no longer panic
- recommend latest kops version in alpha
- recommend latest kops version in stable channel
- kOps managed OIDC provider is no longer needed for IRSA
- set recommended kOps version to 1.23.4 in k8s version 1.18
- Add back missing permissions for legacy CCM. Again.
- Fix CAS cordon flag
- AWS LBC needs ec2:DescribeVpcPeeringConnections for IPv6
- Bump verbosity level for some log statements
- bump aws-cni to version 1.11.4
- hack/update-expected
- remove 'get' from aws-cni clusterRole to reflect https://github.com/aws/amazon-vpc-cni-k8s/pull/2058
- hack/update-expected
- rolling-update: don't deregister our only apiserver
- bump k8s versions in alpha with September releases
- Update dependencies
- Release notes for 1.24.3
- Delete the oldest servers when over the desired count for Hetzner
- [Docs] Fix karpenter link
- Add test for ensuring taints are merged correctly
- Bump stable and alpha channels with latest k8s/kops releases
- Change mock region/subnet names to match mockCloud
- Create edit instancegroup test
- Don't write populated IG state after editing IG
- User IG without image should be allowed
- Bump tests to supported k8s version
- Remove k8s GTE 1.20 checks as it is always true
- Add support for using an existing network for Hetzner
- Update Hetzner CCM to v1.13.0
- Run hack/update-expected.sh
- hetzner: Move out of alpha and drop feature flag
- Release 1.25.0
